docs/user-guides/filtering-configuration.md

```diff
@@ -0,0 +1,186 @@
+# Filter Configuration
+
+This page describes how to enable filtering features on the Vald cluster.
+
+## Requirement
+
+To use any filtering functions with the Vald cluster, you must deploy the ingress and/or egress filter component before deploying the Vald cluster.
+
+The filter component can be deployed anywhere, but it must be able to communicate with the Vald Filter gateway.
+Every filter component should meet Vald's filter gRPC interface.
+
+- The ingress RPC definition
+
+    ```rpc
+    service Filter {
+      // Represent the RPC to generate the vector.
+      rpc GenVector(payload.v1.Object.Blob) returns (payload.v1.Object.Vector) {
+        option (google.api.http) = {
+          post : "/filter/ingress/object"
+          body : "*"
+        };
+      }
+      // Represent the RPC to filter the vector.
+      rpc FilterVector(payload.v1.Object.Vector)
+          returns (payload.v1.Object.Vector) {
+        option (google.api.http) = {
+          post : "/filter/ingress/vector"
+          body : "*"
+        };
+      }
+    }
+    ```
+
+- The egress RPC definition
+
+    ```rpc
+    service Filter {
+      // Represent the RPC to filter the distance.
+      rpc FilterDistance(payload.v1.Object.Distance)
+          returns (payload.v1.Object.Distance) {
+        option (google.api.http) = {
+          post : "/filter/egress/distance"
+          body : "*"
+        };
+      }
+      // Represent the RPC to filter the vector.
+      rpc FilterVector(payload.v1.Object.Vector)
+          returns (payload.v1.Object.Vector) {
+        option (google.api.http) = {
+          post : "/filter/egress/vector"
+          body : "*"
+        };
+      }
+    }
+    ```
+
```

Comment:
should we describe more about difference between Ingress Filter and Egress Filter
 and it's usecase.

Reply:
Could you please tell me more details?
Isn't it enough to show the filter gateway component?